pkg/streamingpromql/engine_test.go

```diff
@@ -1563,15 +1586,15 @@ func TestMemoryConsumptionLimit_SingleQueries(t *testing.T) {
 					span.End()
 
 					if testCase.shouldSucceed {
-						require.NoError(t, res.Err)
-						require.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(rejectedMetrics(0)), "cortex_querier_queries_rejected_total"))
+						assert.NoError(t, res.Err)
```

Comment:
I made these changes so that I can see what assertions are failing on the peak memory limit after the error assertion.